File: csharp/03-cars-assemble/CarsAssemble.cs
using System;

static class AssemblyLine
{
    public static double SuccessRate(int speed)
    {
        if (speed == 0)
        {
            return 0.0;
        }
        else if (speed < 5)
        {
            return 1.0;
        }
        else if (speed < 9)
        {
            return 0.9;
        }
        else if (speed == 9)
        {
            return 0.8;
        }
        else
        {
            return 0.77;
        }
    }

    public static double ProductionRatePerHour(int speed)
    {
        double carProduction = 221;
        double succesRate = SuccessRate(speed);
        double productionRatePerHour = speed * carProduction * succesRate;

        return productionRatePerHour;
    }

    public static int WorkingItemsPerMinute(int speed)
    {
        double productionRatePerHour = ProductionRatePerHour(speed);
        int productionRatePerMinute = (int)productionRatePerHour / 60;

        return productionRatePerMinute;
    }
}

File: java/04-cars-assemble/src/main/java/CarsAssemble.java
public class CarsAssemble {

    private static final int RATE = 221;

    private double getSuccessRate(int speed) {
        if (speed >= 1 && speed <= 4) {
            return 1.0;
        } else if (speed <= 8) {
            return 0.9;
        } else if (speed == 9) {
            return 0.8;
        } else {
            return 0.77;
        }
    }

    public double productionRatePerHour(int speed) {
        double successRate = getSuccessRate(speed);
        return RATE * speed * successRate;
    }

    public int workingItemsPerMinute(int speed) {
        return (int) (productionRatePerHour(speed) / 60);
    }
}
